app/lib/actions/pulp3/content_view_version/export.rb

```diff
@@ -0,0 +1,32 @@
+module Actions
+  module Pulp3
+    module ContentViewVersion
+      class Export < Pulp3::AbstractAsyncTask
+        input_format do
+          param :smart_proxy_id, Integer
+          param :content_view_version_id, Integer
+          param :export_base_dir, String
+        end
+
+        def export(cvv)
+          smart_proxy = SmartProxy.find(input[:smart_proxy_id])
+          api = ::Katello::Pulp3::Api::Core.new(smart_proxy)
+          export_base_dir = input[:export_base_dir]
+          repository_hrefs = cvv.repositories.map { |repo| repo.repository_reference[:repository_href] }
```

Comment:
Do you think it'd be advantageous to pass the repository hrefs as the input to this action? That would help decouple this pulp3 focused code from Katello. We don't use anything else off the CVV

Reply:
I thought about this but I saw code passing in Katello models in pulp3 so decided to be consistent. Fine with passing a list of hrefs if you prefer :)

Reply:
we do typically try to pass models instead of hrefs, as that even more so ties it to pulp3.   But in general it does seem like these actions have more pulp3 logic in them than most others.  In fact most don't call api methods directly, they let the service class do that for them.  Sometimes that is warranted so its not necessary a terrible thing.

I do kinda wonder if the action planning looked more like this:
```
sequence do 
     exporter_output = plan_action(Actions::pulp3::ContentViewVersion::CreateExporter).output
     plan_action(Actions::pulp3::ContentViewVersion::Export, exporter_output[:exporter_href])
     plan_action(::Actions::Pulp3::ContentViewVersion::DestroyExporter, exporter_output[:exporter_href)
end
```

That would allow the actions to be smaller and i think let you push the api calls and logic into an Exporter service class.  We've tried to stick with dumb actions as much as possible and two things help with that:  1) having an action do one and only one things, 2) using the orchestration features of dynflow to string them together.

Reply:
updated